### ISSUE TITLE
"TB hash table size too low!" for antichess tablebases

### ISSUE THREAD

**Issue by ProgramFOX** [ddugovic/Stockfish#558]
I'm loading the following tables from https://tablebase.lichess.ovh/tables/antichess/ using the SyzygyPath option:

* 2-3-4-5-giveaway
* 2-3-4-5-pawnless
* 6-giveaway
* 6-pawnless

And Stockfish errors:

> TB hash table size too low!

**Comment by ProgramFOX** [comment]
Did some changes on tbprobe.cpp, with Niklas' assistance.

* Keeping Size on 1 << 14 but with Overflow 1 << 9 doesn't crash
* Raising Size to 1 << 15 does not crash (Overflow kept at 1)

**Comment by ddugovic** [comment]
Interesting... I'm trying to duplicate (I don't understand this code particularly well right now so I'm unsure what to ask):

```
setoption name SyzygyPath value tb
info string Found 145 tablebases
bench giveaway
...
bench
...
```

**Comment by ProgramFOX** [comment]
It may not be a problem with only 145 tablebase files, but the full set is 8400...

**Comment by ddugovic** [comment]
Ah... I've only got 2-3-4-5 tablebase files, so I'll defer to whatever you think is best...

**Comment by ProgramFOX** [comment]
I have zero clue about the technical details regarding Size and Overflow. All I know is that both of my mentioned changes don't crash, which is equally good in my book, but there's probably a difference...

**Comment by niklasf** [comment]
Putting more than 256 elements into the last hash bucket is a lot. So short of changing the hashtable implementation, I'd recommend going with `Size << 15`. There will be a minor cost for Syzygy lookups in other variants, because the larger table will be less cache efficient.

**Comment by ddugovic** [comment]
OK, a minor cost sounds tolerable (tablebases are quite useful); the first Sygyzy lookup is costly anyway (although reads from disk are much faster than solving endgames).

**Comment by niklasf** [comment]
Looking at the history, we had an even larger hashTable in the past (166bf90e4172b77aa0c420ed271fa55e92301a70). Proposed fix: #559.
